### PR TITLE
bugfix/cop-11847

### PR DESCRIPTION
**Pull request by keenan218**
Fix to replace spaces in pdf title with underscores.

### CODE REVIEW

File: src/pdf/PdfGenerator.ts [hunk 1]
```diff
@@ -28,9 +28,11 @@ export abstract class PdfGenerator {
             const details = data.yourDetails;
             const country = data.country;
             const abbreviated = data.abbreviatedFormName;
-            return submission.data.businessKey ?
+            const submissionTitle = submission.data.businessKey ?
                 `${abbreviated}-${data.businessKey}-${country.nationality}-${details.firstName}${details.lastName}` :
                 `${abbreviated}-${country.nationality}-${details.firstName}${details.lastName}`;
+            const submissionTitleNoSpace = submissionTitle.replace(/\s+/g, '_');
```
Comment (aliktb) [inline src/pdf/PdfGenerator.ts:34]: Just a little note. I see you've done a `/g` global modifier on the regex but there is also a `replaceAll` function within JS that I think could also work and avoids complex regex patterns

Comment (keenan218) [reply src/pdf/PdfGenerator.ts:34]: Okay, ill look into that. just for clarity whats the main issue with doing the global modifier?

Comment (aliktb) [reply src/pdf/PdfGenerator.ts:34]: I don't think it is a big issue doing it this way but I just thought it might be a bit more readable

Comment (keenan218) [reply src/pdf/PdfGenerator.ts:34]: Ah yes, i agree with you there

Comment (keenan218) [reply src/pdf/PdfGenerator.ts:34]: So i had a look into replaceAll and tried to implement it, it seems replaceAll is too new and then brings up a whole load of issues wanting me to update ts to newer version etc

**Review by aliktb — APPROVED** [review]
Approving after discussing on github and offline

**Pull request merged**